framework_patches/nightwatchPatch.js

```diff
@@ -1,29 +1,31 @@
-var path = require('path');
-
 exports.nightwatchPatch = function() {
   var patchOptions = {
     desiredCapabilities: {
       'browserstack.framework': 'nightwatch'
     }
   };
+
   return {
     addCapability: function(key, value) {
       patchOptions['desiredCapabilities'][key] = value;
     },
+
     seleniumHost: function(host, port) {
       patchOptions['seleniumHost'] = host;
       patchOptions['seleniumPort'] = port;
     },
+
     trackFrameworkVersion: function() {
-      var nightwatchPackage = require(path.join(require.main.filename, '../../package.json'));
+      var nightwatchPackage = require(patchOptions.desiredCapabilities['browserstack.framework'] + '/package.json');
```

Comment:
@RobinCK @pulkitsharma07 I think this implementation is better and we should use this